RSA.xs

```diff
@@ -176,18 +176,24 @@ EVP_MD *get_md_bynid(int hash_method)
     }
 }
 #endif
-
 unsigned char* get_message_digest(SV* text_SV, int hash_method)
 {
     STRLEN text_length;
     unsigned char* text;
-
+#if OPENSSL_VERSION_NUMBER >= 0x00908000L
+    unsigned char *md;
+    size_t *mdlen;
+    CHECK_NEW(md, get_digest_length(hash_method), unsigned char);
```

Comment:
This creates a leak in the callers. CHECK_NEW() allocates while the pre-3.0 path relies on libcrypto returning a static buffer and hence doesn't free the returned digest. I think you could simply use a static buffer `unsigned char md[EVP_MAX_MD_SIZE];` here.